### PR TITLE
Pullrequest account deletion webcontroller

### PR DESCRIPTION
**Pull request by BenSartor**

### CODE REVIEW

File: src/test/java/org/simlar/simlarserver/webcontrollers/deleteaccountcontroller/DeleteAccountControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    @Test
+    public void testRenderRequest() throws Exception {
+        mockMvc.perform(get(DeleteAccountController.REQUEST_PATH_REQUEST).accept(MediaType.TEXT_PLAIN))
+                .andExpect(status().isOk())
+                .andExpect(content().string(containsString("Step 1: Request SMS Deletion Code")))
+                .andExpect(content().string(containsString("Phone Number:")))
+                .andExpect(content().string(containsString("<input type=\"submit\" value=\"Request Code\">")));
+    }
+
+    @Test
```
[the remark this answers is not in the document]
Comment (BenSartor) [reply src/test/java/org/simlar/simlarserver/webcontrollers/deleteaccountcontroller/DeleteAccountControllerTest.java]: Thanks @jritzerfeld!

File: src/main/java/org/simlar/simlarserver/webcontrollers/deleteaccountcontroller/XmlErrorExceptionMessage.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+
+    @Getter(AccessLevel.NONE)
+    private final Class<? extends XmlErrorException> exceptionClass;
+
+    private final String message;
+
+    @SuppressWarnings("StaticCollection")
+    private static final Map<Class<? extends XmlErrorException>, XmlErrorExceptionMessage> EXCEPTION_CLIENT_RESPONSE_MAP =
+            Collections.unmodifiableMap(Arrays.stream(values()).collect(Collectors.toMap(response -> response.exceptionClass, response -> response, (class1, class2) -> class1)));
```
[the remark this answers is not in the document]
Comment (BenSartor) [reply src/main/java/org/simlar/simlarserver/webcontrollers/deleteaccountcontroller/XmlErrorExceptionMessage.java]: Using `toUnmodifiableMap` crashes with NPE. I guess null keys are not allowed. All your other recommendations are committed now.

[the remark this answers is not in the document]
Comment (BenSartor) [reply src/main/java/org/simlar/simlarserver/webcontrollers/deleteaccountcontroller/XmlErrorExceptionMessage.java]: This works. But we have to add a null check in the `fromException` function then. Do you really think it gets so much cleaner?

[the remark this answers is not in the document]
Comment (jritzerfeld) [reply src/main/java/org/simlar/simlarserver/webcontrollers/deleteaccountcontroller/XmlErrorExceptionMessage.java]: Yeah. Exactly. null keys or null values only work with some map implementations. So we should leave it as it is.

**Review by jritzerfeld — APPROVED** [review]
You're A Shining Star!

**Pull request merged**